src/jubjub/mod.rs

```diff
@@ -201,17 +201,65 @@ impl JubjubBls12 {
 
         // Create the bases for other parts of the protocol
         {
-            let mut cur = 0;
-            let mut fixed_base_generators = vec![];
+            let mut fixed_base_generators = vec![edwards::Point::zero(); FixedGenerators::Max as usize];
+
+            {
+                // Each generator is found by invoking the group hash
+                // on tag 0x00, 0x01, ... until we find a valid result.
+                let find_first_gh = |personalization| {
+                    let mut cur = 0;
+
+                    loop {
+                        let gh = group_hash::<Bls12>(&[cur], personalization, &tmp);
+                        // We don't want to overflow.
+                        assert!(cur != u8::max_value());
+                        cur += 1;
+
+                        if let Some(gh) = gh {
+                            break gh;
+                        }
+                    }
+                };
+
+                // Written this way for exhaustion (double entendre). There's no
```

Comment:
You can implement into_iter for the enumeration, no?

Reply:
`into_iter` isn't a static method of enums. I don't think there could be a way to use the iterator design to iterate over enums, it would anyway be limited to enums that have simple, irrefutable pattern matches.

Reply:
Oh, maybe you mean I should implement my own `into_iter` for the enum. Maybe, but it's about as complicated as this code.

Reply:
Yup, meant this ^. But you're right, if you only use this once didn't make sense to extract it out.